content/doc_test.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2020 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package content
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"istio.io/istio.io/pkg/test/istioio"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/resource/environment"
+)
+
+var (
+	inst istio.Instance
+
+	testsToRun   = os.Getenv("TEST")
+	testEnv      = os.Getenv("ENV")
+	runAllTests  = (testsToRun == "")
+	testsAsSlice = split(testsToRun)
+
+	setupTemplate = `
+		source "${REPO_ROOT}/content/%v" # snips.sh
+		source "${REPO_ROOT}/tests/util/samples.sh"
+		source "${REPO_ROOT}/tests/util/verify.sh"
+		cd ${REPO_ROOT}
+	`
+
+	snipsFileSuffix = "/snips.sh"
+	testFileSuffix  = "/test.sh"
+	testCleanupSep  = "#! cleanup"
+)
+
+func split(testsAsString string) []string {
+	testsAsSlice := strings.Split(testsAsString, ",")
+	for i := 0; i < len(testsAsSlice); i++ {
```

Comment:
Go has a foreach loop that makes things slightly easier to read: https://golangcode.com/foreach-loops/

Reply:
Changed and it looks better!